plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -627,20 +630,41 @@ type ServiceAffinity struct {
 	labels        []string
 }
 
-func NewServiceAffinityPredicate(podLister algorithm.PodLister, serviceLister algorithm.ServiceLister, nodeInfo NodeInfo, labels []string) algorithm.FitPredicate {
+// serviceAffinityPrecomputation should be run once by the scheduler before looping through the Predicate.  It is a helper function that
+// only should be referenced by NewServiceAffinityPredicate.
+func serviceAffinityPrecomputation(serviceLister algorithm.ServiceLister, podLister algorithm.PodLister) func(pm *predicateMetadata) {
```

Comment:
agree, odd that this doesn't compile for me.` unexpected var, expecting expression` , let me see why